### PR TITLE
fix exitCode type

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?
This PR changes 

- container exitCode type to *int32
-  ecs task health exitCode type to *int32. 

They were *uint32. Agent can error out if exitCode is -1 .

7.53.x backport https://github.com/DataDog/datadog-agent/pull/25770
7.54.x backport https://github.com/DataDog/datadog-agent/pull/25784

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

**Review by mfpierre — APPROVED** [review]

File: comp/core/workloadmeta/collectors/internal/containerd/event_builder_test.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func TestBuildCollectorEvent(t *testing.T) {
 	}
 
 	exitCode := uint32(137)
+	expectedExitCode := int32(exitCode)
 	exitTime := time.Now()
 	fakeExitInfo := &exitInfo{exitCode: &exitCode, exitTS: exitTime}
```
Comment (L3n41c) [inline comp/core/workloadmeta/collectors/internal/containerd/event_builder_test.go:58]: Is this the sign that the type of `exitCode` inside the `exitInfo` struct should also be updated ?

```suggestion
	exitCode := int32(137)
	exitTime := time.Now()
	fakeExitInfo := &exitInfo{exitCode: &exitCode, exitTS: exitTime}
```

**Review by L3n41c — APPROVED** [review]
I don’t know to which extent this may or may not be an issue, but I think that the “real” type for a process exit code is an `int64`.
The `exit` syscall uses C `int`, which is `int64` on 64 bits architectures: https://man7.org/linux/man-pages/man2/exit.2.html
So does the `wait` syscall: https://man7.org/linux/man-pages/man3/wait.3p.html

That’s why the GO docker API to get a container exit code also uses `int64`: https://pkg.go.dev/github.com/docker/docker@v26.1.3+incompatible/api/types/container#WaitResponse
etc.

File: comp/core/workloadmeta/collectors/internal/containerd/event_builder_test.go (hunk 1 above)
Comment (kangyili) [reply comp/core/workloadmeta/collectors/internal/containerd/event_builder_test.go]: https://github.com/DataDog/datadog-agent/pull/25793/commits/6c660d00f1a0c02ed7801650b8f54b994e4ef6a9

**Review by zhuminyi — APPROVED** [review]

**Pull request merged**